### PR TITLE
Update contributing doc

### LINKED ISSUE #1403: Update CONTRIBUTING.md
The current version is outdated in many ways (like the gem name, release procedure..etc.). We should update it to match the project’s current state. And also improve the document with:

- Detailed preparation on patch/minor version releases.
- Explanation on the project structure.
- Additional documents that can empower contributors/users to make new components to their projects.
  - How does exception capturing and tracing features work in integrations like `sentry-rails` or `sentry-sidekiq`?
  - How to write custom tracing subscribers or breadcrumb loggers?

### PR DESCRIPTION
**Pull request by st0012**
It updates the outdated contributing doc that was mainly built for the old SDK. It also describes release preparation in more detail.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,67 +5,71 @@
...
+You can contribute to this project in the following ways:
 
-- File a [bug report] or propose a feature
-- Open a PR for bug fixes or implementing requested features
+- Try out the master branch and provide feedback
+- File a [bug report] or [propose a feature]
+- Open a PR for bug fixes or implement requested features
 - Give feedback to opened issues/pull requests
-- Test the latest version - `gem 'sentry-raven', github: 'getsentry/raven-ruby'`
 - Contribute documentation in the [document repo]
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:16]:
```suggestion
- Contribute documentation in the [sentry-docs repo]
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,67 +5,71 @@
...
-Run:
 
-```bash
-craft prepare x.x.x
-```
+[bug report]: https://github.com/getsentry/sentry-ruby/issues/new?template=bug_report.md
+[propose a feature]: https://github.com/getsentry/sentry-ruby/issues/new?template=feature_request.md
+[extension guideline]: https://github.com/getsentry/sentry-ruby/blob/master/EXTENSION.md
+[SDK structure document]: https://develop.sentry.dev/sdk/unified-api/
+[document repo]: https://github.com/getsentry/sentry-docs
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:73]:
```suggestion
[sentry-docs repo]: https://github.com/getsentry/sentry-docs
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -5,67 +5,71 @@
...
+- File a [bug report] or [propose a feature]
+- Open a PR for bug fixes or implement requested features
 - Give feedback to opened issues/pull requests
-- Test the latest version - `gem 'sentry-raven', github: 'getsentry/raven-ruby'`
 - Contribute documentation in the [document repo]
 
-
 And if you have any questions, please feel free to reach out on [Discord].
 
+## To Contribute Individual Gems
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:20]:
```suggestion
## Contribute To Individual Gems
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -5,67 +5,71 @@
...
-
-## How To Contribute
-
-### Running Tests
-
-#### RAILS_VERSION
+- Use `bundle exec rake` to run tests.
+  - In `sentry-rails`, you can use `RAILS_VERSION=version` to specify the Rails version to test against. Default is `6.1`
+- Use example apps under the `example` or `examples` folder to test the change. (Remember to change the DSN first)
+- To learn more about `sentry-ruby`'s structure, you can read the [SDK structure document]
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:25]:
```suggestion
- To learn more about `sentry-ruby`'s structure, you can read the [Sentry SDK spec]
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -5,67 +5,71 @@
...
 
-Run:
 
-```bash
-craft prepare x.x.x
-```
+[bug report]: https://github.com/getsentry/sentry-ruby/issues/new?template=bug_report.md
+[propose a feature]: https://github.com/getsentry/sentry-ruby/issues/new?template=feature_request.md
+[extension guideline]: https://github.com/getsentry/sentry-ruby/blob/master/EXTENSION.md
+[SDK structure document]: https://develop.sentry.dev/sdk/unified-api/
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:72]:
```suggestion
[Sentry SDK spec]: https://develop.sentry.dev/sdk/unified-api/
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -5,67 +5,71 @@
...
-### Running Tests
-
-#### RAILS_VERSION
+- Use `bundle exec rake` to run tests.
+  - In `sentry-rails`, you can use `RAILS_VERSION=version` to specify the Rails version to test against. Default is `6.1`
+- Use example apps under the `example` or `examples` folder to test the change. (Remember to change the DSN first)
+- To learn more about `sentry-ruby`'s structure, you can read the [SDK structure document]
 
-Because this SDK supports multiple versions of Rails, or even without Rails, you might want to run your test against different versions of Rails.
+## To Write Your Sentry Extension
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:27]:
```suggestion
## Write Your Sentry Extension
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -5,67 +5,71 @@
...
 
-You can do this by changing the `RAILS_VERSION` environment variable:
+Please read the [extension guideline] to learn more. Feel free to open an issue if you find anything missing.
 
+# To Release A Gem
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:31]:
```suggestion
# Release SDK Gem
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -5,67 +5,71 @@
...
-$ bundle exec rake
-```
+## Before the Release
 
-If not specified, it runs tests against `Rails 5.2`. 
+1. Run the example app(s) of the gem and make sure all the events are reported successfully.
+2. Update the changelog's latest `Unreleased` title with the target version.
 
-And if you don't want to run the Rails related test cases, you can use `RAILS_VERSION=0`
+### For minor-version release
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:38]:
```suggestion
### Minor-Version Releases
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -5,67 +5,71 @@
...
 
-```
-$ RAILS_VERSION=0 bundle exec rake # runs without Rails related test cases
-```
+- Make sure all the new features are documented properly in the changelog. This includes but not limited to:
+  - Explanation of the feature.
+  - Sample code for the feature.
+  - Expected changes on the SDK's behavior and/or on the reported events.
+  - Some related screenshots.
+- Prepare a PR in the [document repo] that contains the latest documents.
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:45]:
```suggestion
- Prepare a PR in the [sentry-docs repo] to update relevant content depending on the changes in the new release.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -5,67 +5,71 @@
...
 
-### Testing Your Change Against Example Rails Apps
+### For major-version release
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:47]:
```suggestion
### Major-Version Releases
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -5,67 +5,71 @@
...
-We have a few example apps for different Rails versions under the `/examples` folder. You can use them to perform an end-to-end testing on your changes (just remember to change the DSN to your project's).
+In addition to all the steps listed above, you also need to:
 
-At this moment, we recommend testing against the [Rails 6 example](https://github.com/getsentry/raven-ruby/tree/master/examples/rails-6.0) first. Please read its readme to see what kind of testing you can perform with it.
+- Write a migration guide to
+  - Outline the major changes done in this release.
+  - Explain why upgrading is beneficial.
+  - List all the breaking changes and help users make related changes in their apps.
+- Update gem READMEs.
+- May need to check related wizard files in the [document repo].
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:56]:
```suggestion
- May need to check related wizard files in the [sentry-docs repo].
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -5,67 +5,71 @@
...
 
+## Prepare the Release
 
-## Making a release
+1. Visit the [`Prepare Release`](https://github.com/getsentry/sentry-ruby/actions/workflows/prepare_release.yml) workflow.
+2. Click `Run workflow`.
+3. Fill in the required fields and run the workflow.
 
-Install and use `craft`: https://github.com/getsentry/craft
+### For minor or major-version release
```
Comment (rhcarvalho) [inline CONTRIBUTING.md:64]:
```suggestion
### Extra Steps
```

**Review by rhcarvalho — APPROVED** [review]
Thanks for the update, lgtm! A few optional suggestions.

**Comment by st0012** [comment]
@rhcarvalho thanks for the review. I've applied all the suggestion, merging now 😄

**Pull request merged**